### PR TITLE
[OTX] Replace forked mm libraries with public ones

### PR DESCRIPTION
**Pull request by cih9088**
**I highly encourage you to look into this PR thoroughly as this PR impacts on all tasks in OTX.**

**Changes**
- replaced forked mm libraries with public ones
- employed `mmdeploy` to export model to OpenVINO IR
- enabled multi-node distributed training by extending https://github.com/openvinotoolkit/training_extensions/pull/1392
- used existing hooks (`AdaptiveTrainSchedulingHook`, `EvalHook`) to evaluate model before and after training instead of createing new one https://github.com/openvinotoolkit/training_extensions/pull/1472
- refactored (align code base across cls/det/seg to future refactoring)
- fixed bugs

**Tests**
- all pre-commit tests and pytests are passed

**Related PRs**
- mmdet ([OTX](https://github.com/openvinotoolkit/training_extensions/pull/1355))
- mmseg ([OTX](https://github.com/openvinotoolkit/training_extensions/pull/1400), [MPA](https://github.com/openvinotoolkit/model_preparation_algorithm/pull/96))
- mmcls ([OTX](https://github.com/openvinotoolkit/training_extensions/pull/1435), [MPA](https://github.com/openvinotoolkit/model_preparation_algorithm/pull/106))
- mmdeploy ([OTX](https://github.com/openvinotoolkit/training_extensions/pull/1466), [MPA](https://github.com/openvinotoolkit/model_preparation_algorithm/pull/124))

### CODE REVIEW

**Comment by sungmanc** [comment]
How about merging this PR first and then doing a review, and correction? @harimkang @goodsong81 @wonjuleee @cih9088 
(For sure, before merge, we need to check the CI test results)

Although this PR changes a lot of parts, hard to give a detailed review because it is quite large.
We can continuously edit the overall code after merging process.

**Comment by goodsong81** [comment]
> How about merging this PR first and then doing a review, and correction? @harimkang @goodsong81 @wonjuleee @cih9088 (For sure, before merge, we need to check the CI test results)
> 
> Although this PR changes a lot of parts, hard to give a detailed review because it is quite large. We can continuously edit the overall code after merging process.

I'm fine with that approach. Let's merge this PR after @cih9088 is ready to open this PR as 'ready for review' again and CI is all green.

**Comment by cih9088** [comment]
> I'm fine with that approach. Let's merge this PR after @cih9088 is ready to open this PR as 'ready for review' again and CI is all green.

After tests and pre-commits are passed locally, I will have this ready.

**Comment by sungmanc** [comment]
CI test failed due to the time limit. So, @yunchu  made a PR to change the timeout setting(https://github.com/openvinotoolkit/training_extensions/pull/1513). 

@cih9088, Could you please re-launch the CI test after checking the result of above PR ?
(If there is no CI problem on the above PR, we can re-launch the CI test for this PR)

**Comment by sungmanc** [comment]
> CI test failed due to the time limit. So, @yunchu made a PR to change the timeout setting(#1513).
> 
> @cih9088, Could you please re-launch the CI test after checking the result of above PR ? (If there is no CI problem on the above PR, we can re-launch the CI test for this PR)

The above PR was just merged. Please rebase your branch and do CI tests :)

**Comment by cih9088** [comment]
@harimkang @goodsong81 @wonjuleee @sungmanc 
As we agreed to merge this PR without review and let contributors fix bugs while they are developing on this PR, can we do it without CI result?
CI seems to have a problem with this PR. It gets stuck with no log and reaches to the time limit.

For a reference, here are test results in local with a clean `python 3.8.5` environment installed with `otx/algorithms/init_venv.sh`.
(`pre-commit`, `tests/units`, `tests/integration/cli`, in order)
image

NNCF optimisation for YOLOX failed from time to time.

> AssertionError: trained_performance[k]=0.9512195121951218, evaluated_performance[k]=0.9367088607594937

**Review by JihwanEom — APPROVED** [review]

**Comment by JihwanEom** [comment]
I agree, CI seems like not working properly due to multi-gpu training issues. (Fails after 30 minutes for each TC)

**Comment by sungmanc** [comment]
@cih9088 , What I have experienced during the CI tests, FAIL cases could be shown in CI although all test cases were passed on the local test. So, checking with CI test is needed before merging the branch unless we are in a hurry. 

So, I suggest waiting for the fixing of the CI.

**Review by harimkang — APPROVED** [review]
There is an issue on the HPO side, but we decided to fix it in the next PR, not this PR.

**Review by goodsong81 — APPROVED** [review]
Cheers!

**Comment by sungmanc** [comment]
@cih9088 , maybe we need to change the URL of repo to more public one. What do you think?

image

**Comment by goodsong81** [comment]
> @cih9088 , maybe we need to change the URL of repo to more public one. What do you think?
> 
> image

This PR #1393 might be helpful for reference.

**Comment by cih9088** [comment]
@sungmanc @goodsong81
Agreed. Let me change the repo to the one in #1393 and see how things work out.

File: otx/cli/tools/train.py
Comment (harimkang) [inline otx/cli/tools/train.py:72]: @cih9088  Due to the modification of the code, the otx-workspace (data.yaml created through otx build) cannot be used. (I'm sorry for the confusion because this don't have a proper test case.)
Can I revert this? Will there be any problems elsewhere?

First, I made a PR. Please confirm.
https://github.com/openvinotoolkit/training_extensions/pull/1530

**Comment by sovrasov** [comment]
> @sungmanc @goodsong81 Agreed. Let me change the repo to the one in #1393 and see how things work out.

You may need https://github.com/openvinotoolkit/training_extensions/pull/1486 to use the updated NNCF commit.

**Pull request merged**
